pkg/preparer/setup.go

```diff
@@ -151,22 +151,38 @@ func getTLSClient(certFile, keyFile, caFile string) (*http.Client, error) {
 
 // GetStore constructs a key-value store from the given configuration.
 func (c *PreparerConfig) GetStore() (kp.Store, error) {
-	token, err := loadToken(c.ConsulTokenPath)
+	opts, err := c.getOpts()
 	if err != nil {
 		return nil, err
 	}
+	return kp.NewConsulStore(opts), nil
+}
 
-	client, err := getTLSClient(c.CertFile, c.KeyFile, c.CAFile)
+func (c *PreparerConfig) getOpts() (kp.Options, error) {
+	token, err := loadToken(c.ConsulTokenPath)
 	if err != nil {
-		return nil, err
+		return kp.Options{}, err
+	}
+
+	client, err := c.GetClient()
+	if err != nil {
+		return kp.Options{}, err
 	}
 
-	return kp.NewConsulStore(kp.Options{
+	return kp.Options{
 		Address: c.ConsulAddress,
 		HTTPS:   c.ConsulHttps,
 		Token:   token,
 		Client:  client,
-	}), nil
+	}, nil
+}
+
+func (c *PreparerConfig) GetClient() (*http.Client, error) {
+	client, err := getTLSClient(c.CertFile, c.KeyFile, c.CAFile)
+	if err != nil {
+		return &http.Client{}, err
```

Comment:
In idiomatic Go, the `&http.Client{}` should never be examined. In cases like this, just use `nil`. (The exception being if you return a struct by value, where you _must_ construct an empty value to return.)

After that transform, you can simplify this function further.